### DISCUSSION TITLE
How to keep versions up-to-date?

### DISCUSSION THREAD

**Discussion by amrap030** [techno-tim/k3s-ansible#370] [Questions]
Hello together,

what is the strategy to upgrade k3s, kube-vip etc. to the latest version and keep all versions up-to-date?

Thank you in advance!

**Comment by egandro** [comment]
That is the next to test on my list:

There is an update controller: https://docs.k3s.io/upgrades/automated This will update k3s.

I also thing (sorry - I didn't test it) but just updating the versions and rerunning the role - should work.

(Please test with a test cluster).

**Reply by egandro** [reply]
I have 3 PRs here.

With them you can just update the versions in the `all.yml` file and (re-) run the playbook.

**Reply by CaptainStealthy** [reply]
> I also thing (sorry - I didn't test it) but just updating the versions and rerunning the role - should work.

@egandro Have you had a chance to test this, by any chance? I know that Ansible playbooks are supposed to be idempotent by nature, but I don't want to break my cluster, because I'm still at the "I know enough to be dangerous" stage in my Kubernetes knowledge! 😂

In fact, my cluster is still running v1.23.4!

**Reply by CaptainStealthy** [reply]
@egandro I tested the upgrade controller, and tried to upgrade my cluster from 1.23.4 to 1.23.17 - and it worked, but...now I'm getting constant crash loops for kube-vip and metallb. I'm not sure if any other components are causing issues, but this is all I'm able to see at the moment.

The logs for my kube-vip crashes are the same as in this gist, which I found on a Google search - https://gist.github.com/aceawd/54d25b6839dc894048452244dda642b4

Can you or anyone else help?? :(

**Reply by CaptainStealthy** [reply]
I _think_ I fixed it by removing the kube-vip and metallb files from the server manifest directory, and then I applied the latest kube-vip RBAC manifest from their website, and then _very cautiously_ applied the metallb manifests individually, just once.

I _think_ it's stable now... but I'm probably never upgrading this cluster ever again!!! 😂

**Comment by pbolduc** [comment]
The upstream project, [k3s-io/k3s-ansible](https://github.com/k3s-io/k3s-ansible) has an [upgrade playbook](https://github.com/k3s-io/k3s-ansible/blob/master/playbook/upgrade.yml). It is probably not as robust as as using the update controller. I could see using a upgrade playbook for my lab stuff where I dont care that much and using the update controller for anything prod.

**Comment by egandro** [comment]
My PRs are now in tims branch - I think - you can try just rerunning the installer.. update the versions and be happy.


If course I didn't tried installing from tims Januarry/2023 with the April/2024 version with all levels of k3s versions...
